tests/test_platforms_dummy.py

```diff
@@ -20,3 +21,57 @@ def test_dummy_execute_pulse_sequence():
     sequence = PulseSequence()
     sequence.add(platform.create_qubit_readout_pulse(0, 0))
     result = platform.execute_pulse_sequence(sequence, nshots=100)
+
+
+@pytest.mark.parametrize("parameter", Parameter)
+@pytest.mark.parametrize("average", [True, False])
+@pytest.mark.parametrize("nshots", [100, 1024, 2048])
+def test_dummy_single_sweep(parameter, average, nshots):
+    swept_points = 10
+    platform = Platform("dummy")
+    sequence = PulseSequence()
+    pulse = platform.create_qubit_readout_pulse(qubit=0, start=0)
+    if parameter is Parameter.amplitude:
+        parameter_range = np.random.rand(10)
+    else:
+        parameter_range = np.random.randint(10, size=10)
+    sequence.add(pulse)
+    if parameter in [Parameter.attenuation, Parameter.gain, Parameter.bias]:
+        sweeper = Sweeper(parameter, parameter_range, [pulse.qubit])
+    else:
+        sweeper = Sweeper(parameter, parameter_range, [pulse])
+    results = platform.sweep(sequence, sweeper, average=average, nshots=nshots)
+
+    assert pulse.serial and pulse.qubit in results
+    assert len(results[pulse.serial]) == swept_points if average else int(nshots * swept_points)
+
+
+@pytest.mark.parametrize("parameter1", Parameter)
+@pytest.mark.parametrize("parameter2", Parameter)
+@pytest.mark.parametrize("average", [True, False])
+@pytest.mark.parametrize("nshots", [100, 100])
+def test_dummy_double_sweep(parameter1, parameter2, average, nshots):
+    swept_points = 10
+    platform = Platform("dummy")
+    sequence = PulseSequence()
+    pulse = platform.create_qubit_drive_pulse(qubit=0, start=0, duration=1000)
+    ro_pulse = platform.create_qubit_readout_pulse(qubit=0, start=pulse.finish)
+    sequence.add(pulse)
+    sequence.add(ro_pulse)
+    parameter_range_1 = np.random.rand(10) if parameter1 is Parameter.amplitude else np.random.randint(10, size=10)
+    parameter_range_2 = np.random.rand(10) if parameter2 is Parameter.amplitude else np.random.randint(10, size=10)
```

Comment:
This test is a bit slow, I think we could make it a bit faster. I think we should remove the sleep from the dummy platform (or just set zero sleep time in the runcard) and maybe reduce the number of points in the sweep.

Reply:
Thanks @stavros11. I set the `sleep_time` to zero. Now the test is much faster.